### PR TITLE
Iteration 3 setup

### PR DESCRIPTION
**Pull request by jrwhitmer**
In this PR we have added 6 new classes to our project. We have created them to interact with one another via the sales engine and sales analyst.

### CODE REVIEW

File: lib/customer.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  def initialize(attributes, repository)
+    @id = attributes[:id].to_i
+    @first_name = attributes[:first_name]
+    @last_name = attributes[:last_name]
+    @created_at = Time.parse(attributes[:created_at].to_s)
+    @updated_at = Time.parse(attributes[:updated_at].to_s)
+    @repository = repository
+  end
+end
```
[the remark this answers is not in the document]
Comment (ahenzler) [reply lib/customer.rb:18]: we need to check for this when refactoring.

File: lib/customer_repo.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    @customer_list = customers.map do |customer|
+      Customer.new(customer, self)
+    end
+  end
+
+  def all
+    @customer_list
+  end
+end
```
[the remark this answers is not in the document]
Comment (ahenzler) [reply lib/customer_repo.rb:25]: we all need to keep a eye out for extra whitespace

File: lib/sales_engine.rb [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
 require_relative './invoices_repo'
+require_relative './invoice_item_repo'
+require_relative './transaction_repo'
+require_relative './customer_repo'
 require 'csv'
 
 class SalesEngine
   attr_accessor :items,
                 :merchants,
-                :invoices
```
Comment (ahenzler) [inline lib/sales_engine.rb:9]: good catch on comma

File: spec/sales_analyst_spec.rb [hunk 4]
```diff
@@ -10,29 +10,23 @@
...
-      })
-      sales_analyst = se.analyst
-
       expect(sales_analyst).to be_instance_of(SalesAnalyst)
     end
   end
 
   context 'methods' do
-
-    se = SalesEngine.from_csv({
```
Comment (ahenzler) [inline spec/sales_analyst_spec.rb:28]: smart to pull out sales engine here and just do it once

**Review by ahenzler — APPROVED** [review]
Great setup for iteration 3

File: lib/invoice_item.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+require 'bigdecimal'
```
Comment (deebot10) [inline lib/invoice_item.rb:1]: We need to find out about this missing magic comment

File: lib/invoice_item.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+require 'time'
+
+class InvoiceItem
+
```
[the remark this answers is not in the document]
Comment (deebot10) [reply lib/invoice_item.rb:5]: Look out for white spaces

**Review by deebot10 — APPROVED** [review]

**Review by rcasias — APPROVED** [review]

**Pull request merged**